lib/jarvis/thread_logger.rb

```diff
@@ -0,0 +1,42 @@
+require "cabin"
+
+module Jarvis module ThreadLogger
+  class State
+    def initialize
+      @state = {}
+      @lock = Mutex.new
+    end
+
+    def <<(hash)
+      @lock.synchronize do
+        @state[Thread.current] = hash
```

Comment:
This will cause threads to leak; so maybe having a thing prune through the hash to purge any dead threads?